Backpack/Dialog/Classes/BPKDialogView.m

```diff
@@ -0,0 +1,221 @@
+                                              [self.contentView.trailingAnchor constraintEqualToAnchor:self.contentView.superview.trailingAnchor],
+
+                                              // NOTE: The constant here is explicitly not BPKSpacingLg because the icon view has a white border which
+                                              // blends with the contentView background. Using BPKSpacingLg makes it look unaligned with the BPKSpacingLg below.
+                                              // The border is BPKSpacingMd wide which when added to BPKSpacingBase makes it BPKSpacingLg. This ligns up
+                                              // nicely with the BPKSpacingLg between descriptionLabel and buttonStackView
+                                              [self.titleLabel.topAnchor constraintEqualToAnchor:self.iconView.bottomAnchor constant:BPKSpacingBase],
+                                              [self.titleLabel.centerXAnchor constraintEqualToAnchor:self.titleLabel.superview.centerXAnchor],
+                                              [self.titleLabel.leadingAnchor constraintEqualToAnchor:self.titleLabel.superview.layoutMarginsGuide.leadingAnchor],
+                                              [self.titleLabel.superview.layoutMarginsGuide.trailingAnchor constraintEqualToAnchor:self.titleLabel.trailingAnchor],
+                                              
+                                              [self.descriptionLabel.topAnchor constraintEqualToAnchor:self.titleLabel.bottomAnchor constant:BPKSpacingMd],
+                                              [self.descriptionLabel.leadingAnchor constraintEqualToAnchor:self.descriptionLabel.superview.layoutMarginsGuide.leadingAnchor],
+                                              [self.descriptionLabel.superview.layoutMarginsGuide.trailingAnchor constraintEqualToAnchor:self.descriptionLabel.trailingAnchor],
+                                              
+                                              [self.buttonStackView.topAnchor constraintEqualToAnchor:self.descriptionLabel.bottomAnchor constant:BPKSpacingLg],
+                                              [self.buttonStackView.leadingAnchor constraintEqualToAnchor:self.buttonStackView.superview.layoutMarginsGuide.leadingAnchor constant:(BPKSpacingXl - BPKSpacingLg)],
+                                              [self.buttonStackView.superview.layoutMarginsGuide.trailingAnchor constraintEqualToAnchor:self.buttonStackView.trailingAnchor constant:(BPKSpacingXl - BPKSpacingLg)],
+                                              [self.buttonStackView.superview.layoutMarginsGuide.bottomAnchor constraintEqualToAnchor:self.buttonStackView.bottomAnchor]
+                                              ]];
+}
+
+- (void)layoutSubviews {
+    [super layoutSubviews];
+
+    if (self.hasShadow) {
+        [[BPKShadow shadowLg] applyToLayer:self.layer];
```

Comment:
Have you considered setting shadowPath on the layer (to improve performance)?

https://developer.apple.com/documentation/quartzcore/calayer/1410771-shadowpath
Specifying an explicit path usually improves rendering performance.

Reply:
Not really, @zsbee wrote this code specific code, maybe here tried it?

Reply:
I've updated it to use a computed path for the contentView + iconView